pkg/lib/aws/errors.go

```diff
@@ -175,3 +180,38 @@ func ErrorDashboardHeightOutOfRange(height int) error {
 		Message: fmt.Sprintf("dashboard height %d out of range; height must be between %d and %d", height, _dashboardMinHeightUnits, _dashboardMaxHeightUnits),
 	})
 }
+
+func ErrorNATGatewayLimitExceeded(currentLimit, additionalQuotaRequired int, availabilityZones []string, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNATGatewayLimitExceeded,
+		Message: fmt.Sprintf("NAT gateway limit of %d exceeded in availability zones (%s) of region %s; increase the quota for NAT gateways by at least %d or remove some of the existing ones", currentLimit, s.StrsAnd(availabilityZones), region, additionalQuotaRequired),
+	})
+}
+
+func ErrorNATGatewayLimitExceededInAllAZs(currentLimit, additionalQuotaRequired int, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNATGatewayLimitExceededInAllAZs,
+		Message: fmt.Sprintf("NAT gateway limit of %d exceeded in all availability zones of region %s; increase the quota for NAT gateways by at least %d or remove some of the existing ones", currentLimit, region, additionalQuotaRequired),
+	})
+}
+
+func ErrorEIPLimitExceeded(currentLimit, additionalQuotaRequired int, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrEIPLimitExceeded,
+		Message: fmt.Sprintf("elastic IPs limit of %d exceeded in region %s; increase the quota for elastic IPs by at least %d or remove some of the existing ones", currentLimit, region, additionalQuotaRequired),
+	})
+}
+
+func ErrorInternetGatewayLimitExceeded(currentLimit, additionalQuotaRequired int, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrInternetGatewayLimitExceeded,
+		Message: fmt.Sprintf("internet gateway limit of %d exceeded in region %s; increase the quota for internet gateways by at least %d or remove some of the existing ones", currentLimit, region, additionalQuotaRequired),
+	})
+}
+
+func ErrorVPCLimitExceeded(currentLimit, additionalQuotaRequired int, region string) error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrVPCLimitExceeded,
+		Message: fmt.Sprintf("VPC limit of %d exceeded in region %s; increase the quota for VPCs by at least %d or remove some of the existing ones", currentLimit, region, additionalQuotaRequired),
```

Comment:
nit (for all of them): It might me nice to link to the appropriate page to request a quota increase (e.g. https://console.aws.amazon.com/servicequotas/home?#!/services/ec2/quotas)